File: library/src/main/java/com/hacknife/immersive/NavigationView.java
package com.hacknife.immersive;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;

public class NavigationView extends View {
    private Orientation orientation;

    public NavigationView(Context context) {
        this(context, null);
    }

    public NavigationView(Context context, AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public NavigationView(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        orientation = ImmersiveHelper.getActivityOrientationForContext(context);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        int width;
        int height;
        if (orientation == Orientation._0 || orientation == Orientation._180) {
            width = View.MeasureSpec.getSize(widthMeasureSpec);
            height = ImmersiveHelper.getNavigationBarHeight(getContext());
        } else {
            height = View.MeasureSpec.getSize(heightMeasureSpec);
            width = ImmersiveHelper.getNavigationBarHeight(getContext());
        }
        setMeasuredDimension(width, height);
    }
}

File: library/src/main/java/com/hacknife/immersive/NavigationView.kt
package com.hacknife.immersive

import android.content.Context
import android.util.AttributeSet
import android.util.Log
import android.view.View
import com.hacknife.immersive.ImmersiveHelper.getActivityOrientationForContext
import com.hacknife.immersive.ImmersiveHelper.getNavigationBarHeight

class NavigationView @JvmOverloads constructor(
    context: Context?,
    attrs: AttributeSet? = null,
    defStyleAttr: Int = 0
) : View(context, attrs, defStyleAttr) {
    private val orientation: Orientation by lazy { getActivityOrientationForContext(context!!) }
    private var hide = false
    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val width: Int
        val height: Int
        if (orientation == Orientation._0 || orientation == Orientation._180) {
            width = MeasureSpec.getSize(widthMeasureSpec)
            height = if (hide) 0 else getNavigationBarHeight(context)
        } else {
            height = MeasureSpec.getSize(heightMeasureSpec)
            width = if (hide) 0 else getNavigationBarHeight(context)
        }
        setMeasuredDimension(width, height)
    }

    fun hide() {
        hide = true
    }

}